demo/repo/Jenkinsfile

```diff
@@ -22,6 +22,7 @@ stage('QA') {
 milestone 1
 stage('Staging') {
     lock(resource: 'staging-server', inversePrecedence: true) {
+        milestone 2
```

Comment:
Ok, good catch.

I actually think more natural would be add the milestone after `input` so once a build passes it any older build waiting for `lock` is aborted (without even trying to acquire the lock).

```groovy
milestone 1
stage('Staging') {
    lock(resource: 'staging-server', inversePrecedence: true) {
        node {
            servers.deploy 'staging'
        }
        input message: "Does ${jettyUrl}staging/ look good?"
        milestone 2
    }
    try {
        checkpoint('Before production')
    } catch (NoSuchMethodError _) {
        echo 'Checkpoint feature available in CloudBees Jenkins Enterprise.'
    }
}
```